public/controllers/agent/components/register-agent.js

```diff
@@ -195,12 +196,26 @@ export const RegisterAgent = withErrorBoundary(
     getEnrollDNSConfig = () => {
       let serverAddress = this.configuration['enrollment.dns'] || '';
       this.setState({ defaultServerAddress: serverAddress });
+    };
+
+    getRemoteConfig = async () => {
+      const remoteConfig = await getMasterRemoteConfiguration();
+      let serverAddress = this.defaultServerAddress || this.configuration['enrollment.dns'] || '';
```

Comment:
**thought**: if the `serverAddress` variable is not pretended to be changed, then we could define it as a constant using `const` keyword instead.